pranks/cpu-load.js

```diff
@@ -1,24 +1,47 @@
 const PrankBase = require("./prank-base");
-const sizeOf = require('object-sizeof');
-const fs = require('fs');
-const util = require('util');
-const readFilePromise = util.promisify(fs.readFile);
-const path = require('path');
+const os = require('os');
+const cp = require("child_process");
+const { join } = require("path");
 
 class OverloadCPU extends PrankBase {
-  constructor(expressApp) {
-    super(...arguments);
-  }
-
-  async start() {
-    console.log(`Prank:About to overload CPU`)
-    const fileContent = await readFilePromise(path.join(__dirname, './big-text-file.txt'));
-    require("crypto").createCipher("aes192", fileContent);
-  }
-
-  stop() {
-    //can't stop an atomic operation (crypto)
-  }
+
+	constructor(expressApp) {
+		// @todo, not to do this
+		super(...arguments);
+
+		/**
+		 * @type {number}
+		 * @private
+		 */
+		this.stopTimeout = null;
+
+		/**
+		 * @type {Array<cp.ChildProcess>}
+		 */
+		this.forks = [];
+	}
+
+	/**
+	 * @public
+	 */
+	start(duration = 3000) {
+		this.stop();
```

Comment:
Why is stop called at the stat and not the end here? Isn't `this.forks` still empty here?

Reply:
Minor: Also the orchestrator (chaos control file - responsible for invoking the pranks) is not passing any duration params, it just calls Stop() whenever the schedule says we should stop. A scheduler object allows decoupling the pranks from the timing logic.